internal/indexnode/task.go

```diff
@@ -266,6 +267,12 @@ func (it *indexBuildTask) LoadData(ctx context.Context) error {
 }
 
 func (it *indexBuildTask) BuildIndex(ctx context.Context) error {
+	// support build diskann index
+	indexType := it.newIndexParams["index_type"]
+	if indexType == indexparamcheck.IndexDISKANN {
```

Comment:
a better way is define an interface with 2 implements, one for DiskAnn another for others.

Reply:
The execution paths of all indexes will be unified, and there is no need to create two tasks